helpers/dom.js

```diff
@@ -223,13 +223,25 @@ export function isVisible(node) {
 		if (computedStyle.getPropertyValue('visibility') === 'hidden') return false;
 	}
 
-	const parentNode = getComposedParent(node);
-	if (parentNode) return isVisible(parentNode);
+	if (!parentIsKnownVisible) {
+		const parentNode = getComposedParent(node);
+		if (parentNode) return isVisible(parentNode);
+	}
 
 	return true;
 
 }
 
+export function getFirstVisibleAncestor(node) {
```

Comment:
So this basically returns the node as soon as it no longer finds any hidden ancestors.

It differs from simply `findComposedAncestor(node, n => isVisible(n))` in that `findComposedAncestor(node, n => isVisible(n))` would recurse far more than necessary since both `findComposedAncestor` and `isVisible` are recursive. Whereas the goal of the proposed logic here is to avoid that duplicate tree walking. Or is there another reason?